terraform/bigquery.tf

```diff
@@ -362,6 +461,61 @@ resource "google_bigquery_table" "unique_events_view" {
   ]
 }
 
+# Unique Events -deduplicate rows but as a table function
+resource "google_bigquery_routine" "unique_events_by_date_type" {
+  dataset_id      = google_bigquery_dataset.default.dataset_id
+  routine_id      = "unique_events_by_date_type"
+  definition_body = <<EOT
+    SELECT
+      delivery_id,
+      signature,
+      received,
+      event,
+      payload,
+      JSON_VALUE(payload, "$.organization.login") organization,
+      SAFE_CAST(JSON_VALUE(payload, "$.organization.id") AS INT64) organization_id,
+      JSON_VALUE(payload, "$.repository.full_name") repository_full_name,
+      SAFE_CAST(JSON_QUERY(payload, "$.repository.id") AS INT64) repository_id,
+      JSON_VALUE(payload, "$.repository.name") repository,
+      JSON_VALUE(payload, "$.repository.visibility") repository_visibility,
+      JSON_VALUE(payload, "$.sender.login") sender,
+      SAFE_CAST(JSON_QUERY(payload, "$.sender.id") AS INT64) sender_id,
+    FROM
+      `${google_bigquery_dataset.default.dataset_id}.${google_bigquery_table.raw_events_table.table_id}`
+    WHERE
+      received >= start
+      AND received <= end
+      AND eventTypeFilter == event
```

Comment:
Is `==` valid?

Reply:
You're right, should be =

I hate that SQL is different from every other language in the regard